osbs/build/user_params.py

```diff
@@ -141,61 +145,75 @@ def set_params(self,
-        self.signing_intent = signing_intent
-        self.user = user
-
         if not build_from:
             raise OsbsValidationException('build_from must be defined')
-
         if ':' not in build_from:
             raise OsbsValidationException('build_from must be "source_type:source_value"')
         source_type, source_value = build_from.split(':', 1)
         if source_type not in ('image', 'imagestream'):
             raise OsbsValidationException(
                 'first part in build_from, may be only image or imagestream')
-        if source_type == 'imagestream':
-            self.buildroot_is_imagestream = True
-        self.build_from = build_from
-        self.build_image = source_value
 
         try:
-            self.orchestrator_deadline = int(orchestrator_deadline)
+            orchestrator_deadline = int(build_conf.get_orchestor_deadline())
         except (ValueError, TypeError):
-            self.orchestrator_deadline = ORCHESTRATOR_MAX_RUNTIME
+            orchestrator_deadline = ORCHESTRATOR_MAX_RUNTIME
         try:
-            self.worker_deadline = int(worker_deadline)
+            worker_deadline = int(build_conf.get_worker_deadline())
         except (ValueError, TypeError):
-            self.worker_deadline = WORKER_MAX_RUNTIME
+            worker_deadline = WORKER_MAX_RUNTIME
+
+        # Update kwargs with arguments explicitly accepted by this method
+        kwargs.update({
+            "build_json_dir": build_json_dir,
+            "component": component,
+            "koji_target": koji_target,
+            "koji_task_id": koji_task_id,
+            "platform": platform,
+            "reactor_config_override": reactor_config_override,
+            "signing_intent": signing_intent,
+            "user": user,
+            # Potentially pulled from build_conf
+            "build_from": build_from,
+            "build_image": source_value,
+            "buildroot_is_imagestream": (source_type == "imagestream"),
+            "orchestrator_deadline": orchestrator_deadline,
+            "reactor_config_map": build_conf.get_reactor_config_map(),
+            "scratch": build_conf.get_scratch(scratch),
+            "worker_deadline": worker_deadline,
+        })
+
+        # Drop unknown kwargs, some callers may pass in deprecated params
+        kwargs = {k: v for k, v in kwargs.items() if cls.get_param(k) is not None}
+
+        params = cls(**kwargs)
+        params._populate_image_tag()
+        return params
 
-        self._populate_image_tag()
+    @classmethod
+    def _make_params_super(cls, *args, **kwargs):
+        # Pylint cannot properly infer the return type of an overriden classmethod
+        # that returns cls(). This is an ugly workaround that prevents pylint from
+        # inferring any type at all (thus preventing false-positive warnings).
+        # See https://github.com/PyCQA/pylint/issues/981
+        return BuildCommon.make_params.__func__(cls, *args, **kwargs)
```

Comment:
How do we usually track such workaround?

Just a quick idea, add a `TODO:` for this method to tell "remove this workaround once the pylint issue is fixed"?

Reply:
@MartinBasti usually dislikes TODOs, but in this case it might be valid?

Reply:
TODOs will rot there forever, if you have intention to fix in future open tracking ticket.
I don't think that it's worth tracking.

Reply:
Ok, not adding TODO, not tracking